src/Blocks/Mixins.h

```diff
@@ -164,3 +165,132 @@ class cMetaRotator:
 		return a_Meta;
 	}
 };
+
+
+/** Mixin for blocks whose meta on placement depends on the yaw of the player placing the block. BitMask
+selects the direction bits from the block's meta values. */
+template <class Base, NIBBLETYPE BitMask = 0x7, NIBBLETYPE North = 0x2, NIBBLETYPE East = 0x5, NIBBLETYPE South = 0x3, NIBBLETYPE West = 0x4, bool AssertIfNotMatched = false>
+class cYawRotator:
```

Comment:
Style:
- 5 empty lines between class declarations
- Use 2 digits for hex constants
- The line is quite long, split into separate lines
```suggestion





/** Mixin for blocks whose meta on placement depends on the yaw of the player placing the block.
BitMask selects the direction bits from the block's meta values. */
template <
	class Base,
	NIBBLETYPE BitMask = 0x07,
	NIBBLETYPE North = 0x02,
	NIBBLETYPE East = 0x05,
	NIBBLETYPE South = 0x03,
	NIBBLETYPE West = 0x04,
	bool AssertIfNotMatched = false
>
class cYawRotator:
```

Reply:
I have an objection to using 2-digit hex constants for NIBBLETYPEs, it may be consistent but it's also just _wrong_.

Reply:
I know that they're only 4-bit, but still they use a full byte in memory; I come from the ancient times of byte-sized hex dumps and have a strong dislike of odd-sized hex constants.

I guess we need a third person to decide this tie.

Reply:
http://random.org

Reply:
Not helpful, tiger.

Reply:
cMetaRotator uses 2 digits almost exclusively (apart from the first argument in BlockBed.h for some reason), so in the interests of consistency I would recommend doing the same.

Reply:
All right, tie broken :P